test_prod failing for PyTorch due to unsupported dtypes

Despite the useful updates referenced in this [issue](https://github.com/data-apis/array-api-tests/issues/10), the prod test is still failing [on this line](https://github.com/data-apis/array-api-tests/blob/605621381a48d0373ef40f3f88b217cb817e0404/array_api_tests/test_statistical_functions.py#L147) with a PyTorch backend, even with `FILTER_UNDEFINED_DTYPES = True` and flag `-k "not (uint16 or uint32 or uint64)"`

The error is: `AssertionError: out.dtype=uint8, but should be uint32 [prod(uint8)]`

A PR, which had failing CI as a result, is [here](https://github.com/unifyai/ivy/pull/532).

A short discussion on this topic occurred [here](https://github.com/data-apis/array-api-tests/issues/10#issuecomment-1068469705).

--- a/array_api_tests/test_statistical_functions.py
+++ b/array_api_tests/test_statistical_functions.py
@@ -12,6 +12,7 @@ from . import hypothesis_helpers as hh
 from . import pytest_helpers as ph
 from . import shape_helpers as sh
 from . import xps
+from ._array_module import _UndefinedStub
 from .typing import DataType
 
 pytestmark = pytest.mark.ci
@@ -144,7 +145,10 @@ def test_prod(x, data):
                 _dtype = dh.default_float
     else:
         _dtype = dtype
-    ph.assert_dtype("prod", x.dtype, out.dtype, _dtype)
+    # We ignore asserting the out dtype if what we expect is undefined
+    # See https://github.com/data-apis/array-api-tests/issues/106
+    if not isinstance(_dtype, _UndefinedStub):
+        ph.assert_dtype("prod", x.dtype, out.dtype, _dtype)
     _axes = sh.normalise_axis(kw.get("axis", None), x.ndim)
     ph.assert_keepdimable_shape(
         "prod", out.shape, x.shape, _axes, kw.get("keepdims", False), **kw

